### PR TITLE
added IRQ flood user alert

### PR DESCRIPTION
**Pull request by tridge**

### CODE REVIEW

File: alerts/UA00001.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "dateRaised": 20200929,
+  "affectedFirmware": ["all"],
+  "hardwareLimited": [],
```
Comment (peterbarker) [inline alerts/UA00001.json:4]: This will only affect ChibiOS-based boards.  Maybe wedge it into `description`?

File: alerts/UA00001.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "description": "When using interrupt driven inputs if a hardware failure occurs that generates an extremely high interrupt rate (over 100,000 interrupts per second) the flight controller can become overloaded and cease to function. This affects the following types of inputs: RPM input with RPM_TYPE 1 or 2, Camera feedback with a CAM_FEEDBACK_PIN, Wheel encoders with WENC_TYPE=1, liquid fuel flow monitors with BATT_MONITOR=11 or 12, PWM rangefinder with RFND type 5 or 30, RSSI with RSSI_TYPE=4",
```
Comment (peterbarker) [inline alerts/UA00001.json:5]: RangeFinder type 22 is also affected type 5 is a compatability alias for type 22.

Comment (peterbarker) [reply alerts/UA00001.json:5]: And void `AP_Periph_FW::pwm_hardpoint_init()` looks like it installs a handler which could also be bad.

Also looks like we might also be able to save some bytes by using a `PWMSource` there.

Comment (peterbarker) [reply alerts/UA00001.json:5]: ... and the AP_Radio backends?

... and RCOuput's serial-soft reader?

I *believe* anything doing an `attach_interrupt` may be affected....

File: alerts/UA00001.json (hunk 1 above)
Comment (auturgy) [reply alerts/UA00001.json:4]: STM32 rather than ChibiOS

File: alerts/UA00001.json (hunk 2 above)
Comment (stephendade) [reply alerts/UA00001.json]: Added extra rangefinder and ESC telemetry

File: alerts/UA00001.json (hunk 1 above)
Comment (stephendade) [reply alerts/UA00001.json:4]: Added in STM32

**Review by stephendade — APPROVED** [review]

**Comment by stephendade** [comment]
@tridge and @peterbarker: This is good to go in now - are you ok with the edits we've made?

**Comment by peterbarker** [comment]
On Sun, 4 Oct 2020, Stephen Dade wrote:

> @tridge and @peterbarker: This is good to go in now - are you ok with the edits we've made?

Should we take it to a DevCall?

**Comment by tridge** [comment]
@peterbarker "I believe anything doing an attach_interrupt may be affected...."
no, I don't agree. We're only really vulnerable for certain types of hardware.
A non-DMA UART is also generating interrupts. In theory if a UART misbehaved it could generate 1M interrupts per second and take down the flight controller, but we'd just be worrying users unnecessarily if we warned them about that. Same with AP_Radio - in theory it could generate 1M ISR/sec, but really it won't do that.
The ones we need to warn users about are ones where there is external wiring which is vulnerable to this type of error. Ignition modules are an obvious one.

File: alerts/UA00001.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "dateRaised": 20200929,
+  "affectedFirmware": ["all"],
+  "hardwareLimited": [],
+  "description": "When using interrupt driven inputs on STM32 based boards, if an external hardware failure occurs that generates an extremely high interrupt rate (over 500,000 interrupts per second) the flight controller can become overloaded and cease to function. This affects the following types of inputs: RPM sensor with RPM_TYPE 1 or 2, Camera feedback with a CAM_FEEDBACK_PIN, Wheel encoders with WENC_TYPE=1, liquid fuel flow monitors with BATT_MONITOR=11 or 12, PWM rangefinder with RFND type 5, 22 or 30, RSSI with RSSI_TYPE=4, BRD_RADIO_TYPE value 1, 2 or 3, SERIALx_PROTOCOL with value 16.",
```
Comment (tridge) [inline alerts/UA00001.json:5]: drop BRD_RADIO_TYPE and SERIALx_PROTOCOL

File: alerts/UA00001.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "mitigation": "Disable the parameter associated with the interrupt input",
+  "fixCommit": ["e471b23f5835be122dd461f7efe2a9ac05ee8faf", "965af98f4a2ed12ef76ff0c52d100f8cfad29e20", "c62b86e3c198212212ca84bb991a610a31ae8303"],
+  "dateResolved": null,
+  "linkedIssue": "https://github.com/ArduPilot/ardupilot/issues/15384",
+  "linkedInfo": [],
+  "linkedPR": "https://github.com/ArduPilot/ardupilot/pull/15388",
+  "versionFrom": {},
+  "versionFixed": {},
```
Comment (tridge) [inline alerts/UA00001.json:13]: Copter: 4.0.5, Plane: 4.0.7

File: alerts/UA00001.json (hunk 3 above)
Comment (stephendade) [reply alerts/UA00001.json]: Fixed

File: alerts/UA00001.json (hunk 4 above)
Comment (stephendade) [reply alerts/UA00001.json]: Added

File: alerts/UA00001.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "dateRaised": 20200929,
+  "affectedFirmware": ["all"],
+  "hardwareLimited": [],
+  "description": "When using interrupt driven inputs on STM32 based boards, if an external hardware failure occurs that generates an extremely high interrupt rate (over 500,000 interrupts per second) the flight controller can become overloaded and cease to function. This affects the following types of inputs: RPM sensor with RPM_TYPE 1 or 2, Camera feedback with a CAM_FEEDBACK_PIN, Wheel encoders with WENC_TYPE=1, liquid fuel flow monitors with BATT_MONITOR=11 or 12, PWM rangefinder with RFND type 5, 22 or 30, RSSI with RSSI_TYPE=4.",
+  "mitigation": "Disable the parameter associated with the interrupt input",
+  "fixCommit": ["e471b23f5835be122dd461f7efe2a9ac05ee8faf", "965af98f4a2ed12ef76ff0c52d100f8cfad29e20", "c62b86e3c198212212ca84bb991a610a31ae8303"],
```
Comment (meee1) [inline alerts/UA00001.json:7]: will this git hash always be in the ArduPilot repo? or could it be on another repo/submodule.

Comment (tridge) [reply alerts/UA00001.json:7]: we are linear, so yes, it should always be in our repo, but maybe we should tag UserAlert fixes?

Comment (meee1) [reply alerts/UA00001.json:7]: so a chibios fix would contain the githash of the submodule pointer changing?

Comment (stephendade) [reply alerts/UA00001.json:7]: Yes. For example this commit: https://github.com/ArduPilot/ardupilot/commit/ca167018232651f0eb88c15678deb77f7f964a77 will be represented as ``"fixCommit": ["ca167018232651f0eb88c15678deb77f7f964a77"]``

**Comment by stephendade** [comment]
As per discussion on the devcall, I will squash & merge this in.

**Pull request merged**